flink-runtime/src/test/java/org/apache/flink/runtime/operators/CoGroupTaskTest.java

```diff
@@ -406,46 +328,28 @@ void testCancelCoGroupTaskWhileCoGrouping() {
         getTaskConfig().setDriverPairComparator(RecordPairComparatorFactory.get());
         getTaskConfig().setDriverStrategy(DriverStrategy.CO_GROUP);
 
-        final CoGroupDriver<Record, Record, Record> testTask =
-                new CoGroupDriver<Record, Record, Record>();
+        final CoGroupDriver<Record, Record, Record> testTask = new CoGroupDriver<>();
 
-        try {
-            addInput(new UniformRecordGenerator(keyCnt, valCnt, true));
-            addInput(new UniformRecordGenerator(keyCnt, valCnt, true));
-        } catch (Exception e) {
-            e.printStackTrace();
-            fail("The test caused an exception.");
-        }
+        addInput(new UniformRecordGenerator(keyCnt, valCnt, true));
+        addInput(new UniformRecordGenerator(keyCnt, valCnt, true));
 
-        final AtomicBoolean success = new AtomicBoolean(false);
+        final OneShotLatch delayedLatch = new OneShotLatch();
```

Comment:
```suggestion
        final OneShotLatch delayCoGroupProcessingLatch = new OneShotLatch();
```
nit on the naming: Could we add the step that is supposed to be delayed here.